extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -35,6 +36,58 @@ const BUILT_IN_FUNCTIONS = 'built-in-functions';
  * @const @private {!Object<string, !Object<string, Function>>}
  */
 const FUNCTION_WHITELIST = (function() {
+
+  /**
+   * Custom wrappers for mutating array methods so that we can offer the
+   * same functionality without allowing mutation on variables in bind scope.
+   */
+  const BindArrays = {
+    /**
+     * Similar to Array.prototype.splice, except it returns a copy of the
+     * passed-in array with the desired modifications.
+     * @param {!Array} array
+     * @param {number=} start
+     * @param {number=} deleteCount
+     * @param {...?} items
+     */
+    /*eslint "no-unused-vars": 0*/
+    'copyAndSplice': function(array, start, deleteCount, items) {
+      if (!isArray(array)) {
+        user().warn(
+            TAG,
+            `copyAndSplice: ${array} is not an array; returning null.`);
+        return null;
+      }
+      const copy = array.slice();
+      copy.splice.apply(copy, Array.prototype.slice.call(arguments, 1));
```

Comment:
I thought about that, but it doesn't quite work.

In JS: `[1, 2, 3].splice()` leaves the array unmodified

but with the proposed code you get 

`[undefined, 1, 2, 3]`

since we're calling splice with `[1, 2, 3], undefined, undefined, undefined`

My guess is that slice is also using `arguments` and that explicitly passing it undefined vs having a shorter arguments array results in different behavior.